libraries/Microsoft.Bot.Builder/StreamingRequestHandler.cs

```diff
@@ -0,0 +1,207 @@
+        {
+            this.bot = bot ?? throw new ArgumentNullException(nameof(bot));
+            this.middlewareSet = middlewareSet;
+            userAgent = GetUserAgent();
+            this.onTurnError = onTurnError;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="StreamingRequestHandler"/> class.
+        /// An overload for use with dependency injection via ServiceProvider, as shown
+        /// in DotNet Core Bot Builder samples.
+        /// Throws <see cref="ArgumentNullException"/> if arguments are null.
+        /// </summary>
+        /// <param name="onTurnError">Optional function to perform on turn errors.</param>
+        /// <param name="serviceProvider">The service collection containing the registered IBot type.</param>
+        /// <param name="middlewareSet">An optional set of middleware to register with the bot.</param>
+        public StreamingRequestHandler(Func<ITurnContext, Exception, Task> onTurnError, IServiceProvider serviceProvider, IList<IMiddleware> middlewareSet = null)
+        {
+            services = serviceProvider ?? throw new ArgumentNullException(nameof(serviceProvider));
+            this.middlewareSet = middlewareSet;
+            userAgent = GetUserAgent();
+            this.onTurnError = onTurnError;
+        }
+
+        /// <summary>
+        /// Gets or sets the <see cref="IStreamingTransportServer"/> this handler listens for incoming messages on.
+        /// </summary>
+        /// <value>
+        /// The <see cref="IStreamingTransportServer"/> this handler listens for incoming messages on.
+        /// </value>
+        public IStreamingTransportServer Server { get; set; }
+
+        /// <summary>
+        /// Processes incoming requests and returns the response, if any.
+        /// Checks the validity of the request and invokes the bot.
+        /// </summary>
+        /// <param name="request">A ReceiveRequest from the connected channel.</param>
+        /// <param name="context">Optional context to operate within. Unused in bot implementation.</param>
+        /// <param name="logger">Optional logger used to log request information and error details.</param>
+        /// <returns>A response created by the BotAdapter to be sent to the client that originated the request.</returns>
+        public override async Task<Response> ProcessRequestAsync(ReceiveRequest request, object context = null, ILogger<RequestHandler> logger = null)
+        {
+            var response = new Response();
+
+            try
+            {
+                if (request == null ||
+                    string.IsNullOrEmpty(request.Verb) ||
+                    string.IsNullOrEmpty(request.Path))
+                {
+                    response.StatusCode = (int)HttpStatusCode.BadRequest;
+                    logger?.LogInformation("Request missing verb and/or path.");
```

Comment:
Please move this if block outside the try, and simply return once it's done. 

This whole method is *very* hard to read, as the if/else if / else if / is all nested. 

Please simplify.